finat/point_set.py

```diff
@@ -37,21 +37,21 @@ class PointSingleton(AbstractPointSet):
     """Just a single point."""
 
     def __init__(self, point):
-        point = numpy.asarray(point)
-        assert len(point.shape) == 1
+        point = numpy.asarray(point).reshape(1, -1)
+        assert point.shape[0] == 1
         self.point = point
 
     @property
     def points(self):
-        return self.point.reshape(1, -1)
+        return self.point
 
     @property
     def indices(self):
-        return ()
+        return (gem.Index(extent=len(self.points)),)
```

Comment:
I don't see the point of these changes to `PointSingleton`.  What purpose does this class still have after these changes over just using `PointSet` with a single point?

Reply:
`PointSingleton` is used throughout the code base and often dispatched on or type checked for. I suppose I could go through the code base and see if I can replace it everywhere? Or add a deprecation warning to its `__init__`?

Reply:
Perhaps this ought to return `((),)`?

Reply:
Ah actually that only makes sense if `()` is a valid `gem.Index`. I may have misunderstood what counts as a valid index when I made these changes

Reply:
> `PointSingleton` is used throughout the code base and often dispatched on or type checked for. I suppose I could go through the code base and see if I can replace it everywhere? Or add a deprecation warning to its `__init__`?

This would be a logical next clean-up step, if the changes suggested here to `PointSingleton` were reasonable, which I strongly doubt.